File: bdp-devops/ambari-versions/ambari-3.0.0.0-SNAPSHOT/ambari-common/src/main/python/ambari_commons/exceptions.py
#!/usr/bin/env python

"""
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
"""


class FatalException(Exception):
    def __init__(self, code, reason):
      self.code = code
      self.reason = reason

    def __str__(self):
        return repr("Fatal exception: %s, exit code %s" % (self.reason, self.code))


class NonFatalException(Exception):
  def __init__(self, reason):
    self.reason = reason

  def __str__(self):
    return repr("NonFatal exception: %s" % self.reason)


class TimeoutError(Exception):
  def __str__(self):
    return repr("Timeout error: %s" % self.message)

File: studio/datart-versions/datart-1.0.0-rc.1/core/src/main/java/datart/core/base/exception/Exceptions.java
/*
 * Datart
 * <p>
 * Copyright 2021
 * <p>
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package datart.core.base.exception;

import datart.core.common.MessageResolver;

import java.lang.reflect.Constructor;

public class Exceptions {

    public static void msg(String msg, String... code) {
        tr(BaseException.class, msg, code);
    }

    public static void base(String msg) {
        throw new BaseException(msg);
    }

    public static void notFound(String... msg) {
        tr(NotFoundException.class, "base.not.exists", msg);
    }

    public static void exists(String... msg) {
        tr(ParamException.class, "base.not.exists", msg);
    }

    public static void e(Exception e) {
        throw new BaseException(e);
    }

    public static void tr(Class<? extends BaseException> clz, String messageCode, String... codes) throws RuntimeException {
        BaseException throwable;
        try {
            String message = MessageResolver.getMessages(messageCode, (Object[]) codes);
            Constructor<? extends BaseException> constructor = clz.getConstructor(String.class);
            throwable = constructor.newInstance(message);
        } catch (Exception e) {
            throwable = new BaseException(messageCode);
        }
        throw throwable;
    }


}
